public/js/actions/notifications.js

```diff
@@ -1,14 +1,58 @@
 import * as actionTypes from 'constants/action-types';
+import * as errorCodes from 'constants/error-codes';
 
+import { gettext } from 'utils';
 
-export function addNotification(text, { type='info', userDismissable=false,
-                                autoHide=true } = {}) {
+
+export function addNotification({ text, errorCode,
+                                  type='info', userDismissable,
+                                  autoHide } = {}) {
+
+  if (type === 'error' && !text) {
+    text = gettext('Internal error. Please try again later.');
+  }
+
+  if (type === 'info' && !text) {
+    throw new Error('An info notification requires text');
+  }
+
+  // Default to error type being userDismissable.
+  if (type === 'error' && typeof userDismissable === 'undefined') {
+    userDismissable = true;
+  }
+
+  // Default info type to being hidden on delay.
+  if (type === 'info' && typeof autoHide === 'undefined') {
+    autoHide = true;
+  }
+
+  if (type === 'error' && (!errorCode ||
+                           typeof errorCodes[errorCode] === 'undefined')) {
+    console.error('errorCode', errorCode);
```

Comment:
oh, was this not for debugging? If not, could you preface it with something like 'notifying about error with code...' ?